resteasy/src/main/java/com/cerner/beadledom/resteasy/ModuleProcessor.java

```diff
@@ -0,0 +1,80 @@
+package com.cerner.beadledom.resteasy;
+
+import com.google.inject.Binding;
+import com.google.inject.Injector;
+
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import javax.ws.rs.ext.Provider;
+
+import org.jboss.resteasy.plugins.guice.GuiceResourceFactory;
+import org.jboss.resteasy.plugins.guice.i18n.LogMessages;
+import org.jboss.resteasy.plugins.guice.i18n.Messages;
+import org.jboss.resteasy.spi.Registry;
+import org.jboss.resteasy.spi.ResourceFactory;
+import org.jboss.resteasy.spi.ResteasyProviderFactory;
+import org.jboss.resteasy.util.GetRestful;
+
+/**
+ * Processes a given life cycle injector that contains Guice modules.
+ *
+ * <p>This class expands upon {@link org.jboss.resteasy.plugins.guice.ModuleProcessor} by enforcing
+ * restrictions on available JAX-RS {@link Provider}s to be added to {@link ResteasyProviderFactory}.
+ *
+ * @author Charan Panchagnula
+ */
+public class ModuleProcessor {
```

Comment:
Can this be package private?

Reply:
https://github.com/cerner/beadledom/commit/75d364f068be5b48f12a864aa2a8bd938fb929b8